services/data/src/react/hooks/stableVariablesHash.test.ts

```diff
@@ -0,0 +1,53 @@
+import { stableVariablesHash } from './stableVariablesHash'
+
+describe('stableVariablesHash', () => {
+    it('sorts objects before hashing', () => {
+        const one = {
+            a: {
+                one: 1,
+                two: 2,
+                three: 3,
+            },
+            b: [1, 2, 3],
+            c: 'c',
+        }
+        const two = {
+            c: 'c',
+            b: [1, 2, 3],
+            a: {
+                three: 3,
+                two: 2,
+                one: 1,
+            },
+        }
+
+        expect(stableVariablesHash(one)).toEqual(stableVariablesHash(two))
+    })
+
+    it('can handle primitives', () => {
+        const one = undefined
+        const two = 'string'
+        const three = 3
+        const four = null
+        const five = true
+
+        expect(stableVariablesHash(one)).toMatchInlineSnapshot(`undefined`)
+        expect(stableVariablesHash(two)).toMatchInlineSnapshot(`"\\"string\\""`)
+        expect(stableVariablesHash(three)).toMatchInlineSnapshot(`"3"`)
+        expect(stableVariablesHash(four)).toMatchInlineSnapshot(`"null"`)
+        expect(stableVariablesHash(five)).toMatchInlineSnapshot(`"true"`)
+    })
+
+    it('throws a clear error when something goes wrong', () => {
```

Comment:
```suggestion
    it('throws a clear error when something object contains circular reference', () => {
```

The "circular reference" error will be thrown if anything goes wrong. Right now I can't other things go wrong, so I think we can make this explicit (or analyze the error message and throw an original-error-dependent message)

Reply:
Yeah that makes sense. I'll update it 👍

Reply:
PR with the fix here: https://github.com/dhis2/app-runtime/pull/988

Reply:
Could I get an approval for https://github.com/dhis2/app-runtime/pull/988? Then I'll merge that in and we should be good to go.

Reply:
:heavy_check_mark: